File: solutions/go/sieve/1/sieve.go
package sieve

func Sieve(number int) []int {
	numbers := make([]bool, number+1)
	var primes []int

	for index := 2; index <= number; index++ {
		marked := numbers[index]
		if !marked {
			primes = append(primes, index)
			for secondIndex := 1; ; secondIndex++ {
				currentIndex := index * secondIndex
				if currentIndex > number {
					break
				}
				numbers[currentIndex] = true
			}
		}
	}
	return primes
}

File: solutions/fortran/sieve/1/sieve.f90
module sieve
  implicit none

contains

  function primes(limit) result(array)
    integer, intent(in) :: limit
    integer, allocatable :: array(:)
    logical :: composites(limit)
    integer :: temp(limit), n_primes, index, index2

    n_primes = 0
    composites = .false.

    do index = 2, limit
      if (composites(index)) cycle

      n_primes = n_primes + 1
      temp(n_primes) = index

      do index2 = index ** 2, limit, index
        composites(index2) = .true.
      end do
    end do

    array = temp(1:n_primes)
  end function primes

end module sieve